### ISSUE TITLE
AddAuthenticationJwtBearer should try to get JwtSigningOptions from DI

### ISSUE THREAD

**Issue by hadeedji** [FastEndpoints/FastEndpoints#820]
Currently, AddAuthenticationJwtBearer creates a new JwtSigningOptions then applies the specified action on it. In some situations, this can be difficult. I would like to configure it by adding an options service to the DI, asp.net functions work this way.

In AddAuthenticationJwtBearer, first checking if a JwtSigningOptions is in the container and using that as a base if it is would be nice.

**Comment by dj-nitehawk** [comment]
please explain the exact use case where you need a DI registered `JwtSigningOptions` instance?
would like to understand why you're unable to just do the necessary setup within the `AddAuthenticationJwtBearer()` method?
thanks!

**Comment by hadeedji** [comment]
Hi, really sorry I couldn't get back to you earlier.

Consider the following super simple configuration:
```csharp
builder.Services.AddAuthenticationJwtBearer(s => s.SigningKey = builder.Configuration["jwt:key"]);
```

But I want to make a separate class for JwtOptions, which might include things like expiry time and so on, so I do just that:
```csharp
public class JwtOptions {
    public const string SectionName = "JWT";

    public string Key { get; set; }
    public int Duration { get; set; } = 3600;
}

```

And register it in the DI:
```csharp
builder.Services.AddOptions<JwtOptions>().BindConfiguration(JwtOptions.SectionName)
    .Validate(o => o.Key != null, "no jwt secret key").ValidateOnStart();
builder.Services.AddSingleton(p => p.GetRequiredService<IOptions<JwtOptions>>().Value);

```

Now it would be really nice if in `AddAuthenticationJwtBearer`, I didn't have to dig out the config manually. When I wasn't using FastEndpoints, I could do this:
```csharp
// JwtOptions is in DI container by this point

builder.Services.AddOptions<JwtBearerOptions>(JwtBearerDefaults.AuthenticationScheme)
    .Configure<JwtOptions>((options, myOptions) => {
        // Configure appropriately
    });

builder.Services.AddAuthentication(JwtBearerDefaults.AuthenticationScheme).AddJwtBearer();
```

So following the same approach, I'd really like to do this in FastEndpoints:
```csharp
builder.Services.AddOptions<JwtSigningOptions>()
    .Configure<JwtOptions>((options, myOptions) => options.SigningKey = myOptions.Key);

builder.Services.AddAuthenticationJwtBearer(_ => {});
```

That's my exact use case. Please let me know if something like this is already possible, I'm mainly looking to avoid fetching the options manually, rather let the asp framework get them for me.

Thank you so much, sorry again for the late reply.

**Comment by dj-nitehawk** [comment]
fair enough...
implemented in `v5.31.0.13-beta`
same can be done with `JwtCreationOptions` also now:

https://github.com/FastEndpoints/FastEndpoints/blob/3781099dc775d4683917ad52a722acd8ff86a30b/Src/Library/changelog.md?plain=1#L44-L71
